upload-server/internal/appconfig/appconfig.go

```diff
@@ -276,11 +289,25 @@ func ParseConfig(ctx context.Context) (AppConfig, error) {
 	if err := envconfig.Process(ctx, &ac); err != nil {
 		return AppConfig{}, err
 	} // .if
+
 	if ac.AzureConnection != nil {
-		if ac.AzureConnection.ContainerEndpoint == "" {
+		if ac.AzureConnection.StorageName == "" || ac.AzureConnection.StorageKey == "" {
+			return AppConfig{}, fmt.Errorf("missing required values for connecting to Azure")
+		} else if ac.AzureConnection.ContainerEndpoint == "" {
```

Comment:
```suggestion
		} 
		if ac.AzureConnection.ContainerEndpoint == "" {
```

shouldn't need the else because of the early return.